### PR TITLE
[omega] Remove non-documented "omega with *"

### PR DESCRIPTION
**Pull request by ejgallego**
This form is only used in coq-bignums and not documented. I think
removal is the best choice, specially as `zify` is not part of the
omega plugin anymore.

**Kind:** documentation / removal

- [x] Added / updated test-suite
- [x] Corresponding documentation was added / updated
- [x] Entry added in the changelog

### CODE REVIEW

**Comment by ejgallego** [comment]
After inspecting the situation and the status of `zify` more closely I've opted for removal of this undocumented variant; it seems only bignums will need an overlay.

**Comment by ejgallego** [comment]
Umm, it seems there are no omega maintainers, assigning this to micromega ones then.

**Review by Zimmi48 — APPROVED** [review]
Would be good to have the deprecation merged in 8.11 indeed.

**Review by maximedenes — COMMENTED** [review]
I'm ok with the change if we first deprecate it in 8.11, indeed.

**Comment by ppedrot** [comment]
Can this get rebased and merged relatively quickly for the 8.11 release? Thanks.

**Comment by ejgallego** [comment]
@ppedrot , this is 8.12 as it is the removal; for 8.11 we should add just a deprecation notice + warning if that's to be deprecated.

**Comment by ppedrot** [comment]
Ah, sorry, but then AFAIR this is still not deprecated in 8.11, is it?

**Comment by Zimmi48** [comment]
Indeed, and no PR has yet been opened to deprecate it...

**Comment by ppedrot** [comment]
@Zimmi48 I'll do it, then.

**Comment by ejgallego** [comment]
> Indeed, and no PR has yet been opened to deprecate it...

I'll be happy to do it and try to handle #11304 , on the other hand I'm not sure I understand if @Zimmi48 resumé there is final; also I am a bit afraid to get more involved on this as I was even told "just leave the maintainers do their work" (!)

**Comment by ejgallego** [comment]
I just saw your message @ppedrot , letting you take care of it, here is some stub that should work for the ml part:

https://github.com/coq/coq/pull/11289/files

but I think the suggestion for this particular tactic should be "omega with is deprecated, please use `zify; omega` " or even `zify; lia` ?

**Comment by ppedrot** [comment]
See #11337.

**Comment by Zimmi48** [comment]
@ppedrot has made the deprecation PR #11337 v8.11-only, so I've copied the changelog entry from this PR here (in addition to the changelog entry already there) and it should be merged now.

**Comment by Zimmi48** [comment]
It is my understanding that @maximedenes is not available to merge this PR during Christmas break. It is important not to let this PR wait for too long now that #11337 has been merged, so I'm taking over the assignment and will merge tomorrow barring further comments.

**Comment by ejgallego** [comment]
@Zimmi48 IMHO this PR can wait right? It is a removal for 8.11, the urgent step was the deprecation I think. Anyways Maxime approved this so we could merge.

**Comment by ejgallego** [comment]
Adding a progress tag as we have to include the changelog from
https://github.com/coq/coq/pull/11337

On the other hand it seems to me that a more straightforward path would have been to merge the deprecation first in `master` then backport to 8.11

**Comment by ejgallego** [comment]
Added 8.11 commit.

**Comment by Zimmi48** [comment]
> On the other hand it seems to me that a more straightforward path would have been to merge the deprecation first in `master` then backport to 8.11

Yes, I would have preferred this. I didn't want to be a hassle to @ppedrot, so I accepted the way he did it on the condition that this PR would also include the deprecation changelog entry and be merged now.

> Adding a progress tag as we have to include the changelog from
> #11337

I had already pushed it but you probably didn't realize.

Anyway, there are two options now: merge this PR now (IMHO it is ready and there is no reason to delay it) or open a new PR with the first commit and merge only this one. @ejgallego let me know what path you prefer.

**Comment by ejgallego** [comment]
I guess no point on opening another PR; I think we can move forward with this [if correct of course].

**Pull request merged**
